src/oogeso/dto/oogeso_input_data_objects_multienergy.py

```diff
@@ -0,0 +1,246 @@
+    el_demand_factor: float = None  # electricity demand factor
+    heat_demand_factor: float = None  # heat demand factor
+
+
+@dataclass
+class DeviceSeparator2Data(DeviceData):
+    el_demand_factor: float = None  # electricity demand factor
+    heat_demand_factor: float = None  # heat demand factor
+
+
+@dataclass
+class DeviceStorage_hydrogenData(DeviceData):
+    E_max: float = 0  # MWh storage capacity (maximum stored energy)
+    E_min: float = 0
+    eta: float = 1  # efficiency
+    target_profile: Optional[str] = None  # target profile for use of (seasonal) storage
+    E_cost: float = 0  # cost for depleting storage
+    E_init: float = 0
+
+
+@dataclass
+class DeviceWell_productionData(DeviceData):
+    wellhead_pressure: float = None  # 2 # MPa
+
+
+@dataclass
+class DeviceWell_gasliftData(DeviceData):
+    gas_oil_ratio: float = None  # 500
+    water_cut: float = None  # 0.6
+    f_inj: float = None  # 220 # gas injection rate as fraction of production rate
+    injection_pressure: float = None  # 20 # MPa
+    separator_pressure: float = None  # 2 # MPa
+
+
+@dataclass
+class EdgeHeatData(EdgeData):
+    # Heat loss in MW as function of energy transfer in MW:
+    power_loss_function: Optional[Tuple[List[float], List[float]]] = None
+
+
+@dataclass
+class EdgeHydrogenData(EdgeData):
+    bidirectional: bool = False
+
+
+@dataclass
+class EdgeFluidData(EdgeData):
+    # wellstream, oil, water, gas
+    pressure_from: float = None
+    pressure_to: float = None
+    diameter_mm: float = None
+    temperature_K: float = None
+    height_m: float = 0
+    num_pipes: Optional[int] = None
+    bidirectional: bool = False
+    # allovable relative deviation of pressure from nominal values,
+    pressure_from_maxdeviation: Optional[float] = None
+    pressure_to_maxdeviation: Optional[float] = None
+
+
+@dataclass
+class EdgeGasData(EdgeFluidData):
+    pass
+
+
+@dataclass
+class EdgeOilData(EdgeFluidData):
+    pass
+
+
+@dataclass
+class EdgeWellstreamData(EdgeFluidData):
+    pass
+
+
+@dataclass
+class EdgeWaterData(EdgeFluidData):
+    pass
+
+
+@dataclass
+class CarrierHeatData(CarrierData):
+    pass
+
+
+@dataclass
+class CarrierHydrogenData(CarrierData):
+    energy_value: float = 13  # MJ/Sm3 (calorific value) -> 13 MJ/Sm3
+
+
+@dataclass
+class CarrierGasData(CarrierData):
+    co2_content: float  # kg/Sm3 - see SSB 2016 report -> 2.34 kg/Sm3
+    G_gravity: float  # 0.6
+    Pb_basepressure_MPa: float  # MPa -> 0.101 # MPa
+    R_individual_gas_constant: float  # J/(kg K) -> 500 J/kgK
+    Tb_basetemp_K: float  # K -> 288 K = 15 degC
+    Z_compressibility: float  # 0.9
+    energy_value: float  # MJ/Sm3 (calorific value) -> 40 MJ/Sm3
+    k_heat_capacity_ratio: float  # 1.27
+    rho_density: float  # kg/m3 -> 0.84 kg/m3
+    pressure_method: Optional[str] = "weymouth"  # pressure drop calculation
+
+
+@dataclass
+class CarrierWellstreamData(CarrierData):
+    darcy_friction: float = None  # 0.02
```

Comment:
Are these optional since they are defaulted to None? In some cases this can be Okay, but try to avoid it :)

Reply:
They are not optional, but they need a default value since the base class has optional values (to avoid TypeError: non-default argument '...' follows default argument)